comptests/s3/test_rest_api_rgw.py

```diff
@@ -269,16 +259,331 @@ def test_user_create_36665(self):
+        }
+        self.log.info("Step 7: Deleting Policy by using user Access key and Secret key")
+        status = loop.run_until_complete(
+            self.obj.delete_policy_with_user_keys(user_params3, access_key, secret_key))
+        assert status[0] == HTTPStatus.OK, "Not able to delete policy"
+        self.log.info("Step 8: Successfully delete policy by self")
+        self.created_users.append(user_params)
+        self.log.info("END: %s",test_case_name)
+        self.log.info(
+            "END : test to validate delete policy by using user Access key and secret key")
+
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-41564')
+    def test_41564(self):
+        """Test to delete policy by the user without having caps"""
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Step 1: Create IAM user")
+        user_name = f"{self.user_name_prefix}{str(time.perf_counter_ns()).replace('.', '_')}"
+        email=f"{user_name}@seagate.com"
+        user_params = {
+            'display-name': user_name,
+            'email' : email,
+            'uid' : user_name
+        }
+        self.log.info("Step 2: Started creating IAM user.")
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        self.log.info(user_info)
+        access_key = user_info['keys'][0]['access_key']
+        secret_key = user_info['keys'][0]['secret_key']
+        self.log.info(
+            "Step 3: IAM user is created successfully")
+        policy_document = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Deny",
+                    "Action": "s3:PutObject",
+                    "Resource": "arn:aws:s3:::testbucket/*"
+                }
+            ]
+        }
+        self.log.info(
+            "Step 4: Applying user policy to the user")
+        self.log.info(policy_document)
+        user_params2 = {
+            'UserName' : user_name,
+            'PolicyName' : 'Policy1',
+            'PolicyDocument' : policy_document,
+            'Action' : 'PutUserPolicy',
+            'format' : 'json'
+        }
+        loop = asyncio.get_event_loop()
+        status = loop.run_until_complete(self.obj.put_user_policy(user_params2))
+        self.log.info("Step 5: Successfully Applied user policy")
+        loop = asyncio.get_event_loop()
+        user_params3 = {
+            'UserName' : user_name,
+            'PolicyName' : 'Policy1',
+            'Action' : 'DeleteUserPolicy',
+            'format' : 'json'
+        }
+        status = loop.run_until_complete(
+            self.obj.delete_policy_with_user_keys(user_params3, access_key, secret_key))
+        assert status[0] == HTTPStatus.FORBIDDEN, "Able to delete user policy"
+        self.log.info("Step 6: Validated the deletion of policy by self")
```

Comment:
Please verify policy got deleted by using list user policy/get user policy

Reply:
will update this for all test cases accordingly

Reply:
Here the changes are not required , verifying to delete policy without caps only